### ISSUE TITLE
`less v590` and `less v603` don't print/handle `⋮` character properly

### ISSUE THREAD

**Issue by icy-comet** [gwsw/less#259]
`less` doesn't print `⋮` character properly on Windows 11. I have `v590` installed with `scoop` and I manually compiled `v603-beta` for testing if the issue was fixed.

Here are the relevant screenshots:

![2022_04_02_WindowsTerminal_5](https://user-images.githubusercontent.com/50461557/161375594-f74c2bac-0b15-460e-a6b7-36ad4e0ec2e6.png)

![2022_04_02_WindowsTerminal_6](https://user-images.githubusercontent.com/50461557/161375602-95227cce-ebf5-4927-9a67-fad0217d328a.png)

**Comment by polluks** [comment]
My results:
WSL v603-beta works
PowerShell cat fails
Do you have a hexdump of test.txt? Maybe it's a BOM issue.

**Comment by icy-comet** [comment]
@polluks `less` works fine inside WSL for me too. The problem is only on the Windows side.

Here's the `hexdump` of the text file I used in the screenshots:
```
0000000 8be2 e2ae ae8b 8be2 e2ae ae8b 8be2 e2ae
0000010 ae8b 8be2 e2ae ae8b 8be2 e2ae ae8b 8be2
0000020 e2ae ae8b 8be2 e2ae ae8b 8be2 e2ae ae8b
0000030 8be2 e2ae ae8b 8be2 e2ae ae8b 8be2 e2ae
0000040 ae8b 8be2 e2ae ae8b 8be2 e2ae ae8b 8be2
0000050 e2ae ae8b 8be2 e2ae ae8b 8be2 e2ae ae8b
0000060 8be2 e2ae ae8b 8be2 e2ae ae8b 8be2 e2ae
0000070 ae8b 8be2 00ae
0000075
```

**Comment by polluks2** [comment]
Aha, your file looks strange.

$ xxd mytest.txt 
00000000: e28b ae                                  ...
$ file mytest.txt 
mytest.txt: Unicode text, UTF-8 text

**Comment by polluks** [comment]
Aha, your file looks swapped.

$ xxd mytest.txt
00000000: e28b ae ...
$ file test.txt mytest.txt 
test.txt:   data
mytest.txt: Unicode text, UTF-8 text, with no line terminators

**Comment by icy-comet** [comment]
@polluks I'm sorry I'm not very familiar with encoding and stuff. What does "swapped" mean? Could you also take a look at the linked `deta` issue? Because it's not just that one file having the issue.

**Comment by polluks** [comment]
@icy-comet We have the same bytes, just the order is swapped.
Take a look at U+0800 https://en.wikipedia.org/wiki/UTF-8#Encoding
Your 8be2 seems to be invalid.

**Comment by gwsw** [comment]
@icy-comet , how did you generate the dump? Depending on what command you used, the byte order may or may not be correct. But since the cat command prints the expected data, I'm inclined to think the byte order is not a problem. That is, I think the file contains 39 repetitions of the 3-byte sequence {0xe2,0x8b,0xae} which is UTF-8 for U+22ee VERTICAL ELLIPSIS. The dump output is probably displaying it as little-ending 16-bit words.

Is this the only character that displays incorrectly, or do other Unicode characters also do this? For example, if you download [this file](https://www.w3.org/2001/06/utf-8-test/UTF-8-demo.html) and view it, do the non-ASCII characters display correctly?

**Comment by icy-comet** [comment]
>  how did you generate the dump?

I just used `hexdump` with no options against the text file.

> if you download [this file](https://www.w3.org/2001/06/utf-8-test/UTF-8-demo.html) and view it, do the non-ASCII characters display correctly?

No, at first the characters didn't display correctly. It printed out random characters just like the case with the `⋮` character. Then I set the `LESSCHARSET` environment variable to `utf-8` and it started displaying the Unicode code points like this one `<U+2211>`,  but didn't print the characters. After using `---raw-control-chars` flag, however, all the characters were displayed correctly.

The above flag doesn't work unless the `LESSCHARSET` is defined. Shouldn't just defining the charset be enough to print the characters though? @gwsw

**Comment by gwsw** [comment]
Yes, I would expect setting LESSCHARSET=utf-8 to be sufficient to display UTF-8 characters, and that the --raw-control-chars flag should not be needed. This is how it works on Linux builds -- I'm not sure why it's behaving differently on Windows. Since it's displaying the <U+2211> type of format, it seems that it's thinking that many normal Unicode chars are binary, but I don't see why compiling for Windows should make is_ubin_char behave differently. @adoxa Jason, do you have any ideas?

**Comment by adoxa** [comment]
Explicitly use the UTF-8 code page.

```
PS C:\Projects\less> .\less --version
less 604 (Spencer V8 regular expressions)
Copyright (C) 1984-2022  Mark Nudelman

less comes with NO WARRANTY, to the extent permitted by law.
For information about the terms of redistribution,
see the file named README in the less distribution.
Home page: https://greenwoodsoftware.com/less
PS C:\Projects\less> chcp.com
Active code page: 437
PS C:\Projects\less> .\less -XF .\vell.txt
Γï«Γï«Γï«
PS C:\Projects\less> chcp.com 65001
Active code page: 65001
PS C:\Projects\less> .\less -XF .\vell.txt
⋮⋮⋮
PS C:\Projects\less>
```

My version of `less` uses `lessutf` to handle it automatically.

```
PS C:\Projects\less> echo $env:LESSOPEN
|-lessutf %s
PS C:\Projects\less> chcp.com
Active code page: 437
PS C:\Projects\less> .\less -XF .\vell.txt
Γï«Γï«Γï«
PS C:\Projects\less> less --version
less 600 (PCRE regular expressions)
Copyright (C) 1984-2022  Mark Nudelman
Word wrapping by Jason Hood (http://less.adoxa.vze.com/)

less comes with NO WARRANTY, to the extent permitted by law.
For information about the terms of redistribution,
see the file named README in the less distribution.
Home page: https://greenwoodsoftware.com/less
PS C:\Projects\less> less -XF .\vell.txt
⋮⋮⋮
PS C:\Projects\less>
```

**Comment by icy-comet** [comment]
> Explicitly use the UTF-8 code page.

Unfortunately, even explicitly switching the codepage doesn't seem to have any effect in my case.

![image](https://user-images.githubusercontent.com/50461557/170882530-4454ce0b-24f8-4eb0-a9b4-780a29b917b2.png)

> My version of less uses lessutf to handle it automatically.

What's `lessutf`? I couldn't find anything about it.

**Comment by adoxa** [comment]
> Unfortunately, even explicitly switching the codepage doesn't seem to have any effect in my case.

Sorry, I'm at a loss. Could you try in a `cmd` console?

> What's `lessutf`? I couldn't find anything about it.

Included with my [release](http://less.adoxa.vze.com/).

**Comment by icy-comet** [comment]
> Could you try in a cmd console?

Alright, changing the codepage does work under Command Prompt.

![test2](https://user-images.githubusercontent.com/50461557/170912410-59451e35-016b-45ec-96a8-01bd844a638f.png)

I still wonder why setting just the `LESSCHARSET` var is not enough and why codepage switching has no effect in PowerShell 7.2.4.

**Comment by adoxa** [comment]
Just installed PowerShell 7.2.4 (and updated Windows Terminal Preview to 1.14.1451.0) - it still works for me.
![image](https://user-images.githubusercontent.com/470238/172033271-d647d050-374d-436d-9020-63381cce3d82.png)

**Comment by icy-comet** [comment]
I don't know what changed since my last `less v603` build, but I just rebuilt `v603-beta` with Visual Studio 2022 tools and the problem seems to get resolved as soon as `LESSCHARSET` variable is set now. No need to change the codepage. `v603` also seems to fix the wrong detection of a text file as a binary by `v590`. I tested this in Powershell `v7.2.4` as well as the command prompt (`v10.0.22000.708` if relevant).

The issue persists in `v590` though, which is the current stable version.

![image](https://user-images.githubusercontent.com/50461557/173612771-e89e0027-cdc5-42d0-915f-8ef607005732.png)
